### PR TITLE
Implemented product variations incl. inventory table for variations

### PR DESCRIPTION
**Pull request by SimSync**
**#6 Implemented product variations, incl. inventory table for variations**

This one was a bit bigger.
- Implemented the use of up to 2 variations that will be rendered on the store as select boxes.
- Implemented an "Inventory table" for the product variations, as each variation (/combination) must have it's own inventory. This "table" becomes available in the backend, after the product variations have been selected. To access the table, you have to edit the product again.
- Inventory table will be rendered on the store frontpage as javascript settings to enable a quick check if a product and it's variation combinations are still in stock and to set the status of the "Add to cart" buttons, etc.
- Both shopping carts (menu & page) adjusted to display the selected variation combinations.
- Email templates adjusted to also display the selected variation combinations.

So far i found no bugs, but who knows... ;) 
Would be great if someone could test this extensingly.

grafik

**item_var_info** 
Removed it from admin (still in _sql) as it seams no to be used at any place and (for the moment) i see no use for it.

**Enable/Disable Categories and Products**
Added option to disable single products or a whole category from being displayed in the store.
With that you can disable a product (or category) in case you are editing it or you want to remove a product for a while without having to delete it (and than to re-create it later).
The cart will be checked and disabled items will be removed on display, add and update.

**Item brand**
Implemented item brand on cart by using the category name as item brand name.

### CODE REVIEW

File: vstore_sql.php [hunk 1]
```diff
@@ -6,6 +6,8 @@
   `cart_e107_user` varchar(250) DEFAULT NULL,
   `cart_status` varchar(250) DEFAULT NULL,
   `cart_item` int(11) DEFAULT NULL,
+	`cart_item_var_keys` VARCHAR(100) NOT NULL DEFAULT '',
+	`cart_item_var_values` VARCHAR(255) NOT NULL DEFAULT '',
   `cart_qty` int(11) DEFAULT NULL,
```
Comment (CaMer0n) [inline vstore_sql.php]: If these keys/values work together, perhaps they could be stored in a TEXT field as a JSON array?

Comment (SimSync) [reply vstore_sql.php]: I'll check it.

**Comment by SimSync** [comment]
@CaMer0n Rewrote the cart_item_vars issue to use a single column. But instead of using json, i'm using a "custom" implode, as this makes it easier in my opinion to lookup the values during an sql query.
In addition, i've fixed some typos i've found and the max height of the cart menu.

**Comment by CaMer0n** [comment]
Perfect! Thank you! :+1:

**Pull request merged**